core/src/avm2/globals/string.rs

```diff
@@ -327,6 +362,47 @@ fn replace<'gc>(
     Ok(Value::Undefined)
 }
 
+/// Implements `String.search`
+fn search<'gc>(
+    activation: &mut Activation<'_, 'gc, '_>,
+    this: Option<Object<'gc>>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error> {
+    if let (Some(this), pattern) = (this, args.get(0).unwrap_or(&Value::Undefined)) {
+        let this = Value::from(this).coerce_to_string(activation)?;
+
+        let regexp_class = activation.avm2().classes().regexp;
+        let pattern = if !pattern.is_of_type(activation, regexp_class) {
+            let string = pattern.coerce_to_string(activation)?;
+            regexp_class.construct(activation, &[Value::String(string)])?
+        } else {
+            pattern.coerce_to_object(activation)?
+        };
+
+        if let Some(mut regexp) = pattern.as_regexp_mut(activation.context.gc_context) {
+            let old = regexp.last_index();
+            regexp.set_last_index(0);
+            if let Some(result) = regexp.exec(this) {
+                let found_index = result
+                    .groups()
+                    .next()
+                    .map(|range| range.unwrap_or(0..0).start as i32)
+                    .unwrap_or(0);
```

Comment:
Maybe `.flatten().unwrap_or_default().start` is little clearer?

Reply:
Done